config/cloudinit/datasource/vmware/vmware_amd64.go

```diff
@@ -107,3 +112,41 @@ func urlDownload(url string) ([]byte, error) {
 	client := pkg.NewHTTPClient()
 	return client.GetRetry(url)
 }
+
+func (v VMWare) checkGuestInfo() bool {
+	userData, err := v.FetchUserdata()
+	if err == nil && string(userData) != "" {
+		return true
+	}
+	metadata, err := v.FetchMetadata()
+	if err == nil {
+		if metadata.Hostname != "" {
```

Comment:
These `if conditions..` can be streamlined.
https://rancher.com/docs/os/v1.x/en/installation/running-rancheros/cloud/vmware-esxi/

1. metadata.NetworkConfig.DNS.Nameservers
2. metadata.NetworkConfig.DNS.Search
3. metadata.NetworkConfig.Interfaces
4. metadata.Hostname